cmaes.lua

```diff
@@ -0,0 +1,273 @@
+require 'torch'
+require 'math'
+
+local BestSolution = {} 
+--[[ An implementation of `CMAES` (Covariance Matrix Adaptation Evolution Strategy), 
+ported from https://www.lri.fr/~hansen/barecmaes2.html.
+ 
+Parameters
+----------
+ARGS:
+
+-    `opfunc` : a function that takes a single input (X), the point of 
+          evaluation, and returns f(X) and df/dX. Note that df/dX is not used 
+-    `x` : the initial point
+-    `state.sigma`
+        float, initial step-size (standard deviation in each
+        coordinate)
+-    `state.maxEval`
+        int, maximal number of function evaluations
+-    `state.ftarget`
+        float, target function value
+-    `state.popsize`
+    	 population size. If this is left empty, 
+    	  4 + int(3 * log(|x|)) will be used
+-    `state.ftarget` 
+        stop if fitness < ftarget
+-    `state.verd_disp`
+        int, display on console every verb_disp iteration, 0 for never
+-    `state.args`
```

Comment:
The opfunc usually takes just `x` as the argument.
If some args need to be passed, the opfunc can be a closure.
Would it be OK to remove the state.objfunc_args usage?